src/sas/sascalc/dataloader/loader.py

```diff
@@ -59,98 +53,76 @@ def __init__(self):
         # Register default readers
         readers.read_associations(self)
 
-    def load(self, path, format=None, debug=False):
+    def load(self, path, format=None, use_defaults=True, debug=False):
         """
         Call the loader for the file type of path.
-
         :param path: file path
         :param format: explicit extension, to force the use
             of a particular reader
         :param debug: when True, print the traceback for each loader that fails
-
         Defaults to the ascii (multi-column), cansas XML, and cansas NeXuS
         readers if no reader was registered for the file's extension.
         """
         import traceback
 
         # Gets set to a string if the file has an associated reader that fails
-        msg_from_reader = None
-        try:
-            return super(Registry, self).load(path, format=format)
-        #except Exception: raise  # for debugging, don't use fallback loader
-        except NoKnownLoaderException as nkl_e:
-            if debug: traceback.print_exc()
-            pass  # Try the ASCII reader
-        except FileContentsException as fc_exc:
-            if debug: traceback.print_exc()
-            # File has an associated reader but it failed.
-            # Save the error message to display later, but try the 3 default loaders
-            msg_from_reader = fc_exc.message
-        except Exception:
-            if debug: traceback.print_exc()
-            pass
-
-        # File has no associated reader, or the associated reader failed.
-        # Try the ASCII reader
         try:
-            ascii_loader = ascii_reader.Reader()
-            return ascii_loader.read(path)
-        except NoKnownLoaderException:
+            data_list = super(Registry, self).load(path, format=format)
+            if len(data_list) < 1:
+                raise Exception()
```

Comment:
* Awkward use of exception---you are just using it to enter the exception block.  You can avoid it in this case by restructuring the code (see below).  Also, when debugging it won't display why the exception happened.

* Use `logger.error` instead of `logging.error`.  With `logging` messages are dumped into the global pool and developers won't be able to selectively suppress them (like we did with matplotlib debug messages recently).

* Just use plain `raise` to reraise an exception, not `raise RuntimeError(e)`, unless of course you really intend to change the exception into a RuntimeError exception.

* Instead of a "debug" flag to the method you can use `logger.debug()`.  This change may not be appropriate if the user interface somehow turns on debug and captures stdout to the sasview console.  Maybe hold off for a more complete refactoring?

Revised code to avoid the nested exceptions:
```python

    def load(self, path, format=None, use_defaults=True, debug=False):
        """
        Call the loader for the file type of path.
        :param path: file path
        :param format: explicit extension, to force the use
            of a particular reader
        :param debug: **Deprecated**
        Defaults to the ascii (multi-column), cansas XML, and cansas NeXuS
        readers if no reader was registered for the file's extension.
        """
        import traceback

        # Try reader based on file extension or explicit format.
        try:
            data_list = super(Registry, self).load(path, format=format)
            if data_list:
                return data_list
            if format:
                logger.debug(f"No data returned from '{path}' for format {format}")
            else:
                logger.debug(f"No data returned from '{path}' for the file type")
        except Exception:
            logger.debug(traceback.format_exc())

        # Try backup reader if standard reader fails.
        try:
            return self.load_using_generic_loaders(path)
        except (NoKnownLoaderException, DefaultReaderException) as e:
            logger.debug(traceback.format_exc())
            # If you really want to hide the details of the DefaultReaderException:
            #     logger.error(str(e) if isinstance(e, NonKnownLoaderException)
            #                  else "No default loader can load the data")
            # but the following should be good enough:
            logger.error(str(e))
            # No known reader available. Give up and throw an error
            msg = str(e)
            msg += "\nUnknown data format: {}.\nThe file is not a ".format(path)
            msg += "known format that can be loaded by SasView.\n"
            raise NoKnownLoaderException(msg)
        #except FileContentsException as e: # no special handling needed
        except Exception as e:
            logger.debug(traceback.format_exc())
            raise
```